airbyte-config/config-persistence/src/main/java/io/airbyte/config/persistence/ConfigRepository.java

```diff
@@ -1600,6 +1602,74 @@ public boolean getConnectionHasAlphaOrBetaConnector(final UUID connectionId) thr
     return countResult > 0;
   }
 
+  public Optional<ConnectorBuilderProject> getConnectorBuilderProject(final UUID builderProjectId, final UUID workspaceId) throws IOException {
+    return database.query(ctx -> ctx.select(CONNECTOR_BUILDER_PROJECT.asterisk())
+        .from(CONNECTOR_BUILDER_PROJECT)
+        .where(getBuilderProjectIdCondition(builderProjectId, workspaceId))
+        .fetch())
+        .map(DbConverter::buildConnectorBuilderProject)
+        .stream()
+        .findFirst();
+  }
+
+  public Stream<ConnectorBuilderProject> getConnectorBuilderProjectsByWorkspace(final UUID workspaceId) throws IOException {
+    final Condition matchByIds = CONNECTOR_BUILDER_PROJECT.WORKSPACE_ID.eq(workspaceId);
+
+    return database
+        .query(ctx -> ctx
+            .select(CONNECTOR_BUILDER_PROJECT.ID, CONNECTOR_BUILDER_PROJECT.WORKSPACE_ID, CONNECTOR_BUILDER_PROJECT.NAME,
+                CONNECTOR_BUILDER_PROJECT.ACTOR_DEFINITION_ID)
+            .from(CONNECTOR_BUILDER_PROJECT)
+            .where(matchByIds)
+            .fetch())
+        .map(DbConverter::buildConnectorBuilderProjectWithoutManifestDraft)
+        .stream();
+  }
+
+  public boolean deleteBuilderProject(final UUID builderProjectId, final UUID workspaceId) throws IOException {
+    return database.transaction(ctx -> ctx.deleteFrom(CONNECTOR_BUILDER_PROJECT))
```

Comment:
A thought here - should we actually delete this from DB? It won't leave any record to debug. Normal practice would set a tombstone flag here and avoid querying this when listing/getting builder project. Though for GDPR reasons we need to remember to delete this record if requested.

wdyt?

Reply:
I considered this when setting up the tech spec but ultimately decided against it because there is no strong technical reason that requires it (like other tables referencing it and we don't want to cascade deletes or we need to keep it around for jobs that are still running or something like that).

But if it's unexpected in this case I think that makes it worth implementing it with a soft delete from the start - it will probably be handy at some point in the future. I will adjust the PR.